### ISSUE TITLE
2D Views - Height Labels

### ISSUE THREAD

**Issue by sarahhightower** [CCDevNelson/Snap-Updates#312]
I've noted overlapping height label issues with 2D views below. 

1.  If you have varying heights those heights overlap on the left hand side. (First visual)
2.  If you double pard, but use the same height, the height (sometimes) appears twice and overlaps on the left hand side. (Second visual) In this design, it looks like one height was recognized as 84 and one as 83.25, even though there is a top shelf in all sections. 
3.  The toe kick height overlaps the wall depth line and can make it hard to read (both visuals)        

Would it be possible to resolve these issues by doing the following?

1.   Give each height it's own height line? They can be pretty close together, just not overlapping. 
2.   Prevent the same height from appearing more than once?
3.   Move all of the height labels to the left of the wall depth line?

**Comment by nailsonlandim** [comment]
Hi @sarahhightower @TylerStandage @TedCC1209,

Take a look and tell me what you think.

![312_1](https://user-images.githubusercontent.com/70038700/110807094-74538000-8261-11eb-90b0-575a91bf62d0.png)

![312_2](https://user-images.githubusercontent.com/70038700/110807098-761d4380-8261-11eb-9262-0c82e059e3f8.png)

It's available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23312-height-labels

**Comment by sarahhightower** [comment]
Thanks @nailsonlandim!

This looks great overall. Below are my comments;

1. The pre 2D generation screen looks a little strange with pard sizes in places they don't belong. The measurements are right though (so this isn't the same as issue #313). Not a bit deal...just pointing this out. 

2. Measurements look good separated. Are they intentionally bolded?

![3](https://user-images.githubusercontent.com/72047269/110839692-145cd980-8261-11eb-87c4-9513dd6e1140.jpg)

![4](https://user-images.githubusercontent.com/72047269/110839701-1757ca00-8261-11eb-98e0-d3b640fd7353.jpg)

**Comment by nailsonlandim** [comment]
Hi @sarahhightower, thanks for the feedback.

> 1. The pre 2D generation screen looks a little strange with pard sizes in places they don't belong. The measurements are right though (so this isn't the same as issue #313). Not a bit deal...just pointing this out.

You mean like in this image, right? I switched them back to regular dimensions to deal with a condition that made them as dashed lines. 

![3](https://user-images.githubusercontent.com/72047269/110839692-145cd980-8261-11eb-87c4-9513dd6e1140.jpg)

> 2. Measurements look good separated. Are they intentionally bolded?

They look that way due the use of blender text objects, they are more precise for displacement.
@AntonioGloria tweaked it as much as possible for now. he also worked in making rotation on regular dimensions, but it looked odd in this scenario.

@TylerStandage Also created an Issue to deal with the pards labels overlapping as they require do deal with other conditions conditions. https://github.com/CCDevNelson/Snap-Updates/issues/324/

**Comment by sarahhightower** [comment]
> You mean like in this image, right? I switched them back to regular dimensions to deal with a condition that made them as dashed lines.

Yes, that's the image I was referring to. I wasn't referring to dashed lines though. I was referring to how their placement of the pards was shifted to the left. 

![3](https://user-images.githubusercontent.com/72047269/110846971-96510080-8269-11eb-839a-f4130709c68b.jpg)

> They look that way due the use of blender text objects, they are more precise for displacement.
> @AntonioGloria tweaked it as much as possible for now. he also worked in making rotation on regular dimensions, but it looked odd in this scenario.

Gotcha. Okay thanks.

**Comment by nailsonlandim** [comment]
Got it @sarahhightower, I'll tackle it on #324 Thanks again for the feedback

**Comment by sarahhightower** [comment]
Hi @nailsonlandim. I've been working in this repository today because I need visuals with clear height labels. In doing so, I've noticed that the pard height labels are locked and can not be moved like in other repositories/versions. Is this a change that will implemented when the height labels fix is merged with the master branch?

https://drive.google.com/file/d/1ZGwZmzQTeX7fWOZOd2I1O2jUIw0Y7EGp/view?usp=sharing

**Comment by nailsonlandim** [comment]
Hi @sarahhightower,

We can pair so we can talk more about it.
